.github/workflows/main.yml

```diff
@@ -67,6 +67,8 @@ jobs:
 
   test_no_backends:
     runs-on: ${{ matrix.os }}
+    needs:
+      - check_setup_py
```

Comment:
I suggest to remove this dependency and create another workflow file which would be only triggered if either the setup.py generator script of pyproject.toml changes. An example from arrow: https://github.com/apache/arrow/blob/master/.github/workflows/go.yml#L20-L33

Reply:
done